singularity-eos/eos/modifiers/tilted_maxwell_eos.hpp

```diff
@@ -0,0 +1,314 @@
+
+  using BaseType = T;
+
+  // give me std::format or fmt::format...
+  static std::string EosType() {
+    return std::string("TiltedMaxwellEOS<") + T::EosType() + std::string(">");
+  }
+
+  static std::string EosPyType() { return std::string("Shifted") + T::EosPyType(); }
+
+  // move semantics ensures dynamic memory comes along for the ride
+  TiltedMaxwellEOS(T &&t, Real tilt_fac) : t_(std::forward<T>(t)), tilt_(tilt_fac) {}
+  TiltedMaxwellEOS() = default;
+
+  Real g(Real u) const { return (1.0 - tilt_) * u + tilt_ * u * u;} 
+  Real gp(Real u) const { return 1.0 - tilt_ + 2 * tilt_ * u;} 
+  
+  const Real tilt_;
+  static constexpr Real tol = 1.e-10;
+  static constexpr Real eos_rho_min = 1.e-8;
+  static constexpr Real eos_rho_max = 1.e2;
+
+  auto TestMaxwellConditions(Real pin, Real temp, Real rho_p_min, Real rho_p_max) const { 
+    // Find densities above and below spinodal region with p(rho_l) = pin and p(rho_h) = pin  
+    Real rho_l, rho_h;
+    RootCounts counts_l, counts_h;
+    auto stat_l = regula_falsi([this, temp](Real rho){return this->t_.PressureFromDensityTemperature(rho, temp);}, 
+                               pin, 0.8 * rho_p_max, 
+                               eos_rho_min, rho_p_max,
+                               tol, tol,
+                               rho_l, counts_l);
+    auto stat_h = regula_falsi([this, temp](Real rho){return this->t_.PressureFromDensityTemperature(rho, temp);}, 
+                               pin, 1.1 * rho_p_min, 
+                               rho_p_min, eos_rho_max,
+                               tol, tol,
+                               rho_h, counts_h);
+    // Integrate Maxwell construction integral (i.e. the chemical potential difference) 
+    Real maxwell_int = math_utils::GLQuad32([this, temp, pin](Real rho){return (this->t_.PressureFromDensityTemperature(rho, temp) - pin)/(rho*rho);}, rho_l, rho_h);
+    return std::make_tuple(maxwell_int, rho_l, rho_h);
+  }
+  
+  auto GetMaxwellConstruction(Real temp) const { 
+    constexpr double eps = std::sqrt(std::numeric_limits<Real>::epsilon());
+    
+    // Minimize dPdrho to split isotherm into two regions, if minimum is greater than zero 
+    // there is no spinodal region
+    auto dpdrho = [this, eps, temp](double rho) { 
+      return (this->t_.PressureFromDensityTemperature(rho * (1.0 + eps), temp) 
+      - this->t_.PressureFromDensityTemperature(rho * (1.0 - eps), temp)) / (2 * eps * rho);
+    };
+    auto d2pdrho2 = [this, eps, temp](double rho) { 
+      return (this->t_.PressureFromDensityTemperature(rho * (1.0 + eps), temp) 
+            - 2*this->t_.PressureFromDensityTemperature(rho, temp)
+            + this->t_.PressureFromDensityTemperature(rho * (1.0 - eps), temp)) / (eps * eps * rho * rho);
+    };
+    Real rho_dpdrho_min = 0.0; 
+    RootCounts counts;
+    auto stat = regula_falsi(d2pdrho2, 
+                             0.0, 1.0, 
+                             0.1, 3.0,
```

Comment:
I notice lots of hardcoded numbers floating around. It would be helpful to know why these were chosen to be what they are and maybe have the variables stored as fields in a central location with descriptive names